File: src/main/java/com/hobojoe/enderore/EnderOre.java
package com.hobojoe.enderore;

import com.hobojoe.enderore.block.ModBlocks;
import com.hobojoe.enderore.item.ModItems;
import com.hobojoe.enderore.proxy.CommonProxy;
import com.hobojoe.enderore.recipe.ModRecipes;
import net.minecraftforge.common.config.Configuration;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.SidedProxy;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
import net.minecraftforge.fml.common.registry.GameRegistry;

import java.io.File;

@Mod(modid = EnderOre.MODID, name = EnderOre.name, version = EnderOre.VERSION)
public class EnderOre {
    public static final String MODID = "enderore";
    public static final String name = "Ender Ore";
    public static final String VERSION = "1.0";

    public static Configuration config;


    @Mod.Instance(MODID)
    public static EnderOre instance;

    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent event) {
        System.out.println(name + " is loading!");
        ModItems.init();
        ModBlocks.init();

        File directory = event.getModConfigurationDirectory();
        config = new Configuration(new File(directory.getPath(), "enderore.cfg"));
        Config.readConfig();
    }

    @Mod.EventHandler
    public void init(FMLInitializationEvent event) {
        ModRecipes.init();
    }

    @Mod.EventHandler
    public void postInit(FMLPostInitializationEvent event) {
        if (config.hasChanged()) {
            config.save();
        }
        GameRegistry.registerWorldGenerator(new WorldGen(), 0);
    }

    @SidedProxy(serverSide = "com.hobojoe.enderore.proxy.CommonProxy", clientSide = "com.hobojoe.enderore.proxy.ClientProxy")
    public static CommonProxy proxy;
}

File: src/main/java/com/hobojoe/enderore/EnderOre.kt
package com.hobojoe.enderore

import com.hobojoe.enderore.proxy.CommonProxy
import net.minecraftforge.common.config.Configuration
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.SidedProxy
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
import net.minecraftforge.fml.common.registry.GameRegistry
import java.io.File


@Mod(modid = EnderOre.MODID, modLanguageAdapter = "net.shadowfacts.forgelin.KotlinAdapter")
object EnderOre {

    const val MODID = "enderore"
    const val name = "Ender Ore"
    lateinit var config: Configuration

    @Mod.Instance(MODID)
    lateinit var instance: EnderOre

    @SidedProxy(serverSide = "com.hobojoe.enderore.proxy.CommonProxy", clientSide = "com.hobojoe.enderore.proxy.ClientProxy")
    lateinit var proxy: CommonProxy


    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        println(name + " is loading!")

        val directory = event.modConfigurationDirectory
        config = Configuration(File(directory.path, "enderore.cfg"))
        Config.readConfig()
    }


    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {}


    @Mod.EventHandler
    fun postInit(event: FMLPostInitializationEvent) {
        if (config.hasChanged())
            config.save()
        if (Config.generatesOre)
            GameRegistry.registerWorldGenerator(WorldGen(), 0)
    }
}
